src/Filters/filt.jl

```diff
@@ -99,10 +98,8 @@ end
 # filt! variant that preserves si
 function filt!(out::AbstractArray, f::Biquad{:z}, x::AbstractArray,
                     si::AbstractArray{S,N}=_zerosi(f, x)) where {S,N}
-    ncols = Base.trailingsize(x, 2)
-
     size(x) != size(out) && throw(DimensionMismatch("out size must match x"))
-    (size(si, 1) != 2 || (N > 1 && Base.trailingsize(si, 2) != ncols)) &&
+    (size(si, 1) != 2 || (N > 1 && size(si)[2:end] != size(x)[2:end])) &&
```

Comment:
The checks base on `trailingsize` were more permissive, but the indexing below would then fail if the sizes didn't match, so test up here. (Also, `trailingsize` was accessing a `Base` internal.)